### PR TITLE
add csrf to blocks template

### PR DESCRIPTION
**Pull request by OktayAltay**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -12,6 +12,7 @@
 install_requires = [
     'wagtail>=1.0',
     'django-model-utils>=2.5.2',
+    'django-request-provider==1.0.2',
```
Comment (mikedingjan) [inline setup.py:15]: From the docs: "You shouldn't be using this project. But you should be using the real one which is" https://pypi.python.org/pypi/django-contrib-requestprovider/1.0.1

File: wagtailformblocks/blocks.py [hunk 2]
```diff
@@ -2,9 +2,12 @@
 
 from django import forms
 from django.core.urlresolvers import reverse
+from django.middleware.csrf import get_token
 
 from wagtail.wagtailcore import blocks
```
Comment (mikedingjan) [inline wagtailformblocks/blocks.py:8]: This newline can be removed (ln. 8)

**Review by mikedingjan — CHANGES_REQUESTED** [review]

**Comment by mikedingjan** [comment]
Can you please tell me why this is needed? as far as I know the request is available for blocks from wagtail 1.6 up, so we should just be able to output the csrf_token with the {% csrf_token %} template tag.

**Comment by OktayAltay** [comment]
We are getting csrf_token error when we post the form. That it is missing the token

**Comment by jjanssen** [comment]
@mikedingjan that is not the case. When using the templatetag it doesn't render anything and [this](https://docs.djangoproject.com/en/1.10/ref/csrf/#ajax) is not a reliable situation.

**Comment by mikedingjan** [comment]
I get that, still think that using the template tag from a "request context" should work normally, that request context will be available when using `{% include_block %}` http://docs.wagtail.io/en/v1.7/topics/streamfield.html#template-rendering as far as I can see from the docs it should work like that.

However, if this is not the case a solution should definitely be found, in my opinion implementing a package that says in its own docs `You shouldn't be using this project` isn't the right way to go.

**Comment by tleguijt** [comment]
I agree with Mike.
When I tested it in a project the `csrf_token` was correctly rendered. If it's not, the problem might be that the `RequestContext` is missing?

And probably you don't need the `csrf_token` for this implementation anyway.
One ticket that is still open is #1 . When this is fixed it should be possible to use the form with a normal `POST` action from within the page itself.
Right now the only option is to handle the form submit in javascript and perform an ajax request to the server.
With this you could either;
- read the value that is rendered by `{% csrf_token %}`
- read the CSRF token from the cookie as demonstrated here https://docs.djangoproject.com/en/1.10/ref/csrf/#ajax

We should not introduce another dependency, definitely not one that says `You shouldn't be using this project`

p.s. last commit on the project was 5 years ago

**Comment by OktayAltay** [comment]
Thanks, @mikedingjan i added the {% include_block %} to the template that was rendering my formblock and it worked.

**Pull request closed without merging**